SSR-3683: Develop tooltip and fieldGroup Customization tips (#1495)
kintone-labs/kintone-ui-component: docs/document/docs/guides/in-office-day-list-customization.md

      });
      return tooltipForButton;
    };

  const table = new Kuc.Table({
    label: 'team info',
    columns: [
      {
        title: 'Name',
        field: 'name',
        render: renderName
      },
      {
        title: 'In office day',
        field: 'day',
        render: renderDay
      },
      {
        title: 'Location',
        field: 'location',
        render: renderLocation
      },
      {
        title: 'Note',
        field: 'note',
        render: renderNote
      },
      {
        title: 'Action',
        field: 'action',
        render: renderAction}
    ],
    data: [
      {
        name: '',
        day: [],
        location: '-',
        note: ''
      }
    ]
  });
```

#### Create FieldGroup with Table component
Create a FieldGroup component with the label "KUC team" and the Table component as its content.

```javascript
  const fieldGroup = new Kuc.FieldGroup({
    label: 'KUC team',
    content: table
  });
```

#### Create Tooltip with the FieldGroup
Create Tooltip for the FieldGroup and associated them.

```javascript
  const tooltipForFieldGroup = new Kuc.Tooltip({
    title: 'Office day info',
    container: fieldGroup,
  });
```

> #### Add Event listener for FieldGroup 

Reviewer: Please delete the trailing space.

```suggestion
#### Add Event listener for FieldGroup
```

---
id: in-office-day-list-customization
title: In-office day list customization
sidebar_label: In-office day list customization
---

## Overview
This article explains how to utilize and customize the FieldGroup and Tooltip components in the context of creating an In-Office Day List.<br/>
In this context, a Table component within the FieldGroup allows team members to specify which days they will be in the office.<br/>

We assume the following scenario:

> 1. Basic Usage for FieldGroup and Tooltip (*FieldGroup and Tooltip*)
> 2. Append Table component to FieldGroup content (*FieldGroup*)
> 3. Using Two Types of Tooltip (*Tooltip*)

Reviewer: ```suggestion
1. Basic Usage for FieldGroup and Tooltip
2. Append Table component to FieldGroup content
3. Using Two Types of Tooltip
```

### Using Two Types of Tooltip

> You may have noticed that when creating a Tooltip for a button, I will introduce a new property `describeChild`.

Reviewer: Please add link of Tooltip describeChild explanation part.

This property is set to false by default.<br/>
You can effectively improve the accessibility and usability of your application with the `describeChild` property.<br/>

> Now, let's enable the [screen reader](https://support.apple.com/en-us/guide/voiceover/vo2682/mac) and try moving the focus to the button to see what the screen reader reads.<br/>

Reviewer: I think it will be better to use MDN screen reader link rather than the specific screen reader: VoiceOver page.

```suggestion
Now, let's enable the [screen reader](https://developer.mozilla.org/en-US/docs/Glossary/Screen_reader) and try moving the focus to the button to see what the screen reader reads.<br/>
```

#### Tooltip represents the purpose of the element

> If the Tooltip represents the purpose of the element itself (provides a label for the element it is attached to). You need to set `describeChild` to false.

Reviewer: ```suggestion
If the Tooltip represents the purpose of the element itself (provides a label for the element it is attached to), you need to set `describeChild` to false.
```

```javascript
  const tooltipForButton = new Kuc.Tooltip({

>     title: 'Submit changes or additions for this entry.',

Reviewer: This is description or explanation of the target container, so describeChild should be set to true.
So this is not a good example for describeChild: false case.
Please consider another one.

  });
```

> ![render](/img/tooltip_describeChild_false.gif)

Reviewer: This image has poor resolution compared with the second gif image.
Could you please improve that? I can see the vertical and horizontal light lines in the image.

Reviewer: Please check this again!

In the examples, the screen reader replaces the button's original content "Submit" with the Tooltip's content ("Submit changes or additions for this entry") when the button gains focus.<br/>
This is because when `describeChild` is false, Tooltip adds [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) to its child elements (button, FieldGroup), causing the screen reader to read the content of [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) instead of the original content.<br/>

#### Tooltip represents the description of the element

> If the Tooltip represents the description of the element (provides additional information or a supplementary description about the element it is attached to).
> You need to set `describeChild` to true.

Reviewer: ```suggestion
If the Tooltip represents the description of the element (provides additional information or a supplementary description about the element it is attached to), you need to set `describeChild` to true.
```
